src/azure-cli-core/azure/cli/core/telemetry.py

```diff
@@ -151,6 +152,7 @@ def _get_base_properties(self):
             'Context.Default.VS.Core.Distro.Name': _get_distro_name(),  # eg. 'CentOS Linux 8'
             'Context.Default.VS.Core.Distro.Id': _get_distro_id(),  # eg. 'centos'
             'Context.Default.VS.Core.Distro.Version': _get_distro_version(),  # eg. '8.4.2105'
+            'Context.Dafault.VS.Core.Istty': sys.stdin.isatty(),
```

Comment:
I didn't import the func from `knack` but use the [code logic](https://github.com/microsoft/knack/blob/e0c14114aea5e4416c70a77623e403773aba73a8/knack/prompting.py#L25-L28) directly:
```
def verify_is_a_tty():
    if not sys.stdin.isatty():
        logger.debug('No tty available.')
        raise NoTTYException()
```